### PR TITLE
Re-enable Personalization click metrics

### PR DESCRIPTION
**Pull request by ninaceban**
## Description

The customer will be able to disable or enable the link click tracking, this configuration will not affect Personalization click tracking metrics.

I have added `event.documentMayUnload()` so that in case the link click or metric is set on a exit link the request is triggered before navigation.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/Personalization/createOnClickHandler.js [hunk 1]
```diff
@@ -41,6 +41,8 @@ export default ({
             }
           };
         }
+        // TODO: move this in attachClickActivityCollector get link details here and determine whether should set use beacon
+        event.documentMayUnload(); // this is needed in case it is an exit link and but clickCollection is disabled
```
Comment (jonsnyder) [inline src/components/Personalization/createOnClickHandler.js:45]: I think its okay to just use /collect no matter what type of link it is.

File: src/components/ActivityCollector/configValidators.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ governing permissions and limitations under the License.
 import { string, boolean } from "../../utils/validation";
 
 export default {
-  clickCollectionEnabled: boolean().default(true),
+  clickCollectionEnabled: boolean().default(false),
```
Comment (jonsnyder) [inline src/components/ActivityCollector/configValidators.js:16]: I don't think we want to change the default yet. Changing the default is a breaking change. I documented a strategy for changing the default here: https://wiki.corp.adobe.com/display/DMSArchitecture/Migration+plan+to+collect+less+by+default
I think making the changes from the wiki should be a separate ticket/PR.

**Review by jonsnyder — COMMENTED** [review]
I was expecting an updated functional test. It looks like we don't have a test for an interact with a proposition. You should add one.

File: src/components/Personalization/createOnClickHandler.js (hunk 1 above)
Comment (ninaceban) [reply src/components/Personalization/createOnClickHandler.js:45]: ok, thank you

File: src/components/ActivityCollector/configValidators.js (hunk 2 above)
Comment (ninaceban) [reply src/components/ActivityCollector/configValidators.js:16]: I see, ok, I'll do that in a separate PR. Thank you Jon for referencing the Wiki

File: src/components/ActivityCollector/attachClickActivityCollector.js [hunk 3]
```diff
@@ -17,6 +17,7 @@ const createClickHandler = ({ eventManager, lifecycle, handleError }) => {
     // TODO: Consider safeguarding from the same object being clicked multiple times in rapid succession?
     const clickedElement = clickEvent.target;
     const event = eventManager.createEvent();
+    event.documentMayUnload();
```
Comment (jfkhoury) [inline src/components/ActivityCollector/attachClickActivityCollector.js:20]: Can we add an inline comment explaining why we have added this expression?

**Review by jfkhoury — APPROVED** [review]
Can we please update the description of this PR to remove the disabling of the config?

good work Nina!

File: src/components/ActivityCollector/attachClickActivityCollector.js (hunk 3 above)
Comment (ninaceban) [reply src/components/ActivityCollector/attachClickActivityCollector.js:20]: sure!

**Review by jonsnyder — APPROVED** [review]

**Pull request merged**